test/queue_test.rb

```diff
@@ -129,6 +128,30 @@ def enqueue(method, *args)
     QC.default_queue = nil
   end
 
+  def test_multi_threaded_server
+    adapter1 = QC::ConnAdapter.new
+    adapter2 = QC::ConnAdapter.new
+    q1 = q2 = nil
+
+    QC.default_conn_adapter = adapter1
+
+    t1 = Thread.new do
+      QC.default_conn_adapter = adapter1
+      q1 = QC::Queue.new('queue1').conn_adapter
+    end
+
+    t2 = Thread.new do
+      QC.default_conn_adapter = adapter2
+      sleep 0.01
```

Comment:
tests pass on my machine without this sleep. What are we waiting for with this call?